File: sic-api/Entities/Su/SuChatGroupLog.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using sic_api.Entities;

namespace sic_api.Entities.Su;

[Table("su_chat_group_log")]
public class SuChatGroupLog : BaseBusinessEntity
{
    [Required]
    [Column("group_id")]
    public Guid GroupId { get; set; }

    [Required]
    [MaxLength(100)]
    [Column("sender_id")]
    public string SenderId { get; set; } = default!;

    [Required]
    [MaxLength(4000)]
    [Column("message")]
    public string Message { get; set; } = string.Empty;

    [Required]
    [Column("message_type")]
    public ChatMessageType MessageType { get; set; } = ChatMessageType.Text;

    [Column("attachment_id")]
    public Guid? AttachmentId { get; set; }

    [Required]
    [Column("is_cancelled")]
    public bool IsCancelled { get; set; } = false;

    [Column("cancelled_at")]
    public DateTime? CancelledAt { get; set; }

    [MaxLength(100)]
    [Column("cancelled_by")]
    public string? CancelledBy { get; set; }

    public SuChatGroup? Group { get; set; }
    public SuUpload? Attachment { get; set; }
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/su/SuChatGroupLog.java
package com.softinter.sicapi.entity.su;

import com.softinter.sicapi.entity.base.BaseBusinessEntity;
import com.softinter.sicapi.entity.enums.ChatMessageType;

import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "su_chat_group_log")
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class SuChatGroupLog extends BaseBusinessEntity {

    @Column(name = "group_id", insertable = false, updatable = false)
    private UUID groupId;

    @Column(name = "sender_id", nullable = false, length = 100)
    private String senderId;

    @Column(name = "sender_name", length = 100)  
    private String senderName;

    @Column(name = "message", nullable = false, length = 4000)
    private String message = "";

   
    @Column(name = "message_type", nullable = false)
    private ChatMessageType messageType = ChatMessageType.TEXT;

    @Column(name = "attachment_id", insertable = false, updatable = false)
    private UUID attachmentId;

    @Column(name = "is_cancelled", nullable = false)
    private Boolean isCancelled = false;

    @Column(name = "cancelled_at")
    private Instant cancelledAt;

    @Column(name = "cancelled_by", length = 100)
    private String cancelledBy;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "group_id", nullable = false)
    private SuChatGroup group;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "attachment_id")
    private SuUpload attachment;

}